### PR TITLE
Refactor Robyn and SubRouter classes

### PR DESCRIPTION
**Pull request by dave42w**
Refactor both Robyn and SubRouter to be cleaner and leaner without changing the developer API at all.
Use unit tests to confirm the existing behaviour continues to work unchanged.

## Description

This PR fixes https://github.com/sparckles/Robyn/issues/1071

## Summary

This PR does the following:

- added tests to create instances of Robyn and SubRouter with empty routing tables. This confirms that the doc examples still work.
- added a test of the app.directory_path (I was having problems with test_cli, this is to catch any regression)
- split the two big classes Robyn and SubRouter into a new module robyn_app.py

## PR Checklist

Please ensure that:

- [X] The PR contains a descriptive title
- [X] The PR contains a descriptive summary of the changes
- [X] You build and test your changes before submitting a PR.
- [X] You have added relevant documentation
- [X] You have added relevant tests. We prefer integration tests wherever possible

## Pre-Commit Instructions:

- [X] Ensure that you have run the [pre-commit hooks](https://github.com/sparckles/robyn#%EF%B8%8F-to-develop-locally) on your PR.

### CODE REVIEW

**Comment by dave42w** [comment]
I'm not going to actually refactor the two classes until we get this basic move into a new module merged (it's going to create conflicts with other PRs if we don't get these merged in small bits.

File: robyn/robyn_app.py
Comment (sansyrox) [inline robyn/robyn_app.py:1]: Hey @dave42w 👋 

Thank you for the PR but I am not a big fan of moving these classes outside the main `__init__.py` file. I'd want the `Robyn` class to be present in the main file only.

Comment (dave42w) [reply robyn/robyn_app.py:1]: Hi,
I'd be interested to understand better the reasons.
Is that just Robyn?
If I create a RobynBase class for Robyn and SubRouter where would that be best to go?
Thanks

Comment (dave42w) [reply robyn/robyn_app.py:1]: Hi @sansyrox 
Can I have some direction on this. There is some neat refactoring possible to simplify Robyn and SubRouter and reduce potential issues. But I'm not sure how you want this to be organised.

It seems messy to me to have all this in _init__.py but also odd to move RobynBase elsewhere while leaving Robyn behind.

But I'm willing to go in steps that you are happy with.

**Pull request closed without merging**